src/viper/trans.ml

```diff
@@ -759,9 +759,36 @@ and exp ctxt e =
      !!(match e.it with
         | M.TupE es -> CallE (tag.it, List.map (exp ctxt) es)
         | _ -> CallE (tag.it, [exp ctxt e]))
+  | M.CallE ({ it = M.DotE (_, { it = "forall" | "exists" as predicate_name; _ }); _ }, _inst, { it = M.FuncE (_, _, _, pattern, _, _, e); note; _ }) ->
+    let binders = extract_binders pattern in
+    let typs =
+      match M.(note.note_typ) with
+      | T.Func (_, _, _, [ T.Tup args ], _) -> args
+      | T.Func (_, _, _, [ arg ], _) -> [ arg ]
+      | _ -> []
+    in
+    let typed_binders = zip_with_exn binders typs ~f:(fun b t ->
+      let b = id b in
+      b, t)
+    in
+    let ctxt = add_locals ctxt typed_binders in
+    let typed_binders = List.map (fun (b, t) -> (b, tr_typ t)) typed_binders in
+    let e = exp ctxt e in
```

Comment:
```suggestion
    let ctxt = add_locals ctxt typed_binders in
    let typed_binders = List.map (fun (b, t) -> (b, tr_typ t)) typed_binders in
    let e = exp ctxt e in
```
```suggestion
    let ctxt' = add_locals ctxt typed_binders in
    let typed_binders = List.map (fun (b, t) -> (b, tr_typ t)) typed_binders in
    let e = exp ctxt' e in
```

Reply:
Do we want to live without shadowing? These ticks look annoying sometimes IMHO